src/hooks/useNotificationModal.ts

```diff
@@ -220,13 +227,22 @@ export const useNotificationModal = ({
       owner_type: "person",
       regex_selectors: Object.entries(regexSelectorInputs)
         .filter((v) => v[1])
-        .map(([type, data]) => ({ type, data })),
+        .map(([t, data]) => ({ type: t, data })),
     };
   };
 
   const disableAddCriteria =
     regexSelectorPropsTemplate.length >= (regexSelectors?.length ?? 0);
 
+  const onChangeSubscriptionMethod = (v: string) => {
+    setSelectedSubscriptionMethod(v);
+    Cookies.set("subscription-method", v, { expires: 365 });
+  };
+
+  const onChangeTrigger = (v: number) => {
+    setSelectedTriggerIndex(v);
+    Cookies.set(`${type}-notification-trigger`, `${v}`, { expires: 365 });
```

Comment:
Just a thought that is maybe out of scope — setting a cookie based on the trigger index could mess up the cookies if we ever make any updates to the dropdown by adding or reordering (which should theoretically be a low-risk update). There are also lots of duplicate key errors on this dropdown because the key is based on the `payloadResourceIdKey`, which is a duplicate value for many entries [here](https://github.com/evergreen-ci/spruce/blob/main/src/components/PatchActionButtons/addNotification/PatchNotificationModal.tsx#L91). It would be nice to use the [`trigger`](https://github.com/evergreen-ci/spruce/blob/bc200b7a93d9e311e0e24603b718c8da02e4b821/src/hooks/useNotificationModal.ts#L279) field instead for the key and value for each Option, and then use that value to store the cookie here.

But that is a bigger refactor, and I think this would work for the foreseeable future! We could make a tech debt ticket instead.

Reply:
Yeah def I actually tried to take a crack at refactoring the state management in this hook to support this originally. It ended up becoming a larger undertaking than i wanted to pursue. I opened a ticket to come back to this in the future. https://jira.mongodb.org/browse/EVG-15458 
i wasn't aware about the duplicate key errors. I will fix those though.